src/controllers/TwitterController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Twitter-specific endpoints live here
+ */
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Subscriber\Oauth\Oauth1;
+
+class TwitterController extends ApiController {
+    public function handle(Request $request, $db) {
+        // really need to not require this to be declared
+    }
+
+    public function getRequestToken($request, $db){
+        // only trusted clients can change account details
+        $clientId = $request->getParameter('client_id');
+        $clientSecret = $request->getParameter('client_secret');
+        $this->oauthModel = $request->getOauthModel($db);
+        if (!$this->oauthModel->isClientPermittedPasswordGrant($clientId, $clientSecret)) {
+            throw new Exception("This client cannot perform this action", 403);
+        }
+
+        // API call to twitter, get request token
+        $client = new Client([
+            'base_url' => 'https://api.twitter.com/',
+            'defaults' => ['auth' => 'oauth']
+
+        ]);
+
+        $oauth = new Oauth1([
+            'consumer_key'    => $this->config['twitter']['consumer_key'],
+            'consumer_secret' => $this->config['twitter']['consumer_secret'],
+        ]);
+        $client->getEmitter()->attach($oauth);
+
+        $res = $client->post('oauth/request_token');
```

Comment:
Guzzle throws an exception on error. I'm not sure we should be exposing it directly to our client. Thoughts?